JavadocStyleGuide.md

```diff
@@ -0,0 +1,242 @@
+# Gradle Javadoc Style Guide
+
+## 1.1 Formatting
+
+### 1.1.1 General form
+
+The basic formatting of Javadoc blocks is as seen in this example:
+
+```java
+/**
+ * Returns an Image object that can then be painted on the screen.
+ * <p>
+ * The url argument must specify an absolute {@link URL}. 
+ * The name argument is a specifier that is relative to the url argument.
+ * This method always returns immediately, whether or not the image exists. 
+ *
+ * @param url an absolute URL giving the base location of the image
+ * @param name the location of the image, relative to the url argument
+ * @return the image at the specified URL
+ * @see Image
+ */
+public Image getImage(URL url, String name) {
+   try {
+       return getImage(new URL(url, name));
+   } catch (MalformedURLException e) {
+       return null;
+   }
+}
+```
+
+Basic formatting rules:
+
+- The first line contains the begin-comment delimiter ( `/**`).
+- The first sentence is a summary.
+- Notice the inline tag `{@link URL}`, which converts to an HTML hyperlink pointing to the documentation for the URL class.
+- If you have more than one paragraph in the doc comment, separate the paragraphs with a `<p>` paragraph tag, as shown.
+- Insert a blank comment line between the description and the list of tags, as shown.
+- The first line that begins with an `@` character ends the description; you cannot continue the description following block tags.
+- Block tags must be added in order.
+- The last line contains the end-comment delimiter ( `*/`).
+
+So lines won't wrap, limit any doc-comment lines to 80 characters.
+
+### 1.1.2 HTML
+
+Javadoc permits only a subset of HTML tags: `"a", "abbr", "acronym", "address", "area", "b", "bdo", "big", "blockquote", "br", "caption", "cite", "code", "colgroup", "dd", "del", "dfn", "div", "dl", "dt", "em", "fieldset", "font", "h1", "h2", "h3", "h4", "h5", "h6", "hr", "i", "img", "ins", "kbd", "li", "ol", "p", "pre", "q", "samp", "small", "span", "strong", "sub", "sup", "table", "tbody", "td", "tfoot", "th", "thead", "tr", "tt", "u", "ul", "var"`.
+
+Attempt to limit your usage of HTML to: `"h1", "h2", "h3", "h4", "em", "li", "ol", "p", "ul"`.
```

Comment:
```suggestion
Attempt to limit your usage of HTML to: `"a", "h1", "h2", "h3", "h4", "em", "li", "ol", "p", "ul"`.
```

Also, I propose that if you're including 4 levels of headers in a single javadoc comment, you're doing it wrong.

Reply:
Sigh...lol.